test/compilable/issue19243.sh

```diff
@@ -0,0 +1,63 @@
+#! /usr/bin/env bash
+
+# bypassing this test:
+#  - on windows
+#  - on FreeBSD 32 bits
+#       test fails looling for liborig.so; don't know why but shouldn't block fixing all other platforms
+#  - on Circle CI with no_pic. (need PIC to run the test)
+if [[ $OS = *"win"* ]]; then exit 0; fi
+if [[ $OS = *"freebsd"* ]] && [[ $MODEL = *"32"* ]]; then exit 0; fi
+if [ ! -z ${PIC-} ]; then
+    if [ ${PIC} -eq "0" ]; then
+        exit 0
+    fi
+fi
+
+SOEXT=.so
+if [[ $OS = "osx" ]]; then
+    SOEXT=.dylib
+fi
```

Comment:
Might make sense to add this to the globally available environment variables (https://github.com/dlang/dmd/blob/master/test/tools/exported_vars.sh).
Documentation is at the end of test/README.md